Bottom: b4c822dd74377c5db236f6e504859a77e1b98205
Top:    a32334e7c7b624a17584a5528f65008b8e626d63
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-13 01:57:53 +0100

Fix some broken assumptions regarding 32-bit integer coordinates in polygon1.c


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 667e26d..566596d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2510,8 +2510,8 @@ poly_IniContour (PLINE * c)
     return;
   /* bzero (c, sizeof(PLINE)); */
   c->head.next = c->head.prev = &c->head;
-  c->xmin = c->ymin = 0x7fffffff;
-  c->xmax = c->ymax = 0x80000000;
+  c->xmin = c->ymin = COORD_MAX;
+  c->xmax = c->ymax = -COORD_MAX - 1;
   c->is_round = FALSE;
   c->cx = 0;
   c->cy = 0;
@@ -2899,7 +2899,7 @@ poly_InsideContour (PLINE * c, Vector p)
   info.f = 0;
   info.p[0] = ray.X1 = p[0];
   info.p[1] = ray.Y1 = p[1];
-  ray.X2 = 0x7fffffff;
+  ray.X2 = COORD_MAX;
   ray.Y2 = p[1] + 1;
   if (setjmp (info.env) == 0)
     r_search (c->tree, &ray, NULL, crossing, &info);
